File: kd/java/ex3.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class ex3 {
    public static void main(String[] args) {
        HashMap<String,Integer> map=new HashMap<>();
        try {
            Scanner sc = new Scanner(new File("data.csv"));

            while (sc.hasNextLine()) {
                String line = sc.nextLine();      

                String[] parts = line.split(","); 

                for (String value : parts) {
                    map.put(value,map.getOrDefault(value, 0)+1);
                }
            }

            sc.close();

        } catch (FileNotFoundException e) {
            System.out.println("File not found!");
        }

        map.entrySet()
           .stream()
           .sorted((a, b) -> b.getValue() - a.getValue())   
           .limit(3)
           .forEach(entry -> 
                System.out.println(entry.getKey() + " = " + entry.getValue()));
    }
}

File: python/ex3.py
f= open("c:/Users/Admin/Desktop/kd/python/data.csv","r")

map={}
for i in f:
    parts= i.strip().split(",")
    for j in parts:
        map[j]=map.get(j,0)+1

top3 = sorted(map.items(), key=lambda x: x[1], reverse=True)[:3]
print(top3)

f.close()